collector/corosync/corosync.go

```diff
@@ -36,129 +35,70 @@ type corosyncCollector struct {
 	collector.DefaultCollector
 	cfgToolPath    string
 	quorumToolPath string
+	cfgToolParser  Parser
 }
 
 func (c *corosyncCollector) Collect(ch chan<- prometheus.Metric) {
 	log.Debugln("Collecting corosync metrics...")
 
-	err := c.collectRingErrorsTotal(ch)
-	if err != nil {
-		log.Warnf("Corosync Collector scrape failed: %s", err)
-	}
+	// We suppress the exec errors because if any interface is faulty the tools will exit with code 1, but we still want to parse the output.
```

Comment:
why we need to parse a broken output instead of returning error before?

Reply:
this logic was already present: we want to still parse the output in case of failure because we need to report faulty rings.

Reply:
ok